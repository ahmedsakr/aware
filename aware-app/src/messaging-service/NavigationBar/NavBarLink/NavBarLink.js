import React, {Component} from 'react';
import './NavBarLink.scss';

class NavBarLink extends Component {
  render() {
    const {className, icon, name} = this.props;

    return (
      <div class={className + " navigation-menu-link"}>
        <span class={icon} aria-hidden="true"></span>
        <p>{name}</p>
      </div>
    );
  }
}

export default NavBarLink;

class NavBarLogo extends Component {
  render() {
    const {className} = this.props;

    return (
      <div class={className + " navigation-menu-logo"}>
        <p>AWARE</p>
      </div>
    );
  }
}

class NavBarTitle extends Component {
  render() {

    const {className, title} = this.props;

    return (
      <div class={className + " navigation-menu-title"}>
        <p>{title}</p>
      </div>
    );
  }
}

export {
  NavBarLogo,
  NavBarTitle
}