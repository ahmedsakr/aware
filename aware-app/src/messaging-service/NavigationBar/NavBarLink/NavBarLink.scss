/*
 * NavBarLink styling
 *
 * Defines layout fundamentals for NavBar Links.
 */

.navigation-menu-title {
    color: #ffc000;
}

.navigation-menu-logo>p {
    color: #babdc0;
    display: inline;
    font-size: 1.5em;
}

.navigation-menu-link:hover {
    background-color: #18202c;
    text-decoration: none;
}

.navigation-menu-link>p {
    color: #babdc0;
    display: inline;
    padding-left: 5px;
    font-size: 1em;
    text-align: center;
}

.navigation-menu-link>span {
    display: inline;
    color: #babdc0;
}

@media screen and (min-width: 576px) and (max-width: 768px) {
    .navigation-menu-title>p {
        font-size: 0.8em;
    }

    .navigation-menu-link {
        font-size: 0.75em;
    }
}

@media screen and (min-width: 768px) and (max-width: 992px) {
    .navigation-menu-title>p {
        font-size: 1em;
    }

    .navigation-menu-link {
        font-size: 0.75em;
    }
}

@media screen and (min-width: 992px) and (max-width: 1200px) {
    .navigation-menu-title>p {
        font-size: 1.5em;
    }

    .navigation-menu-link {
        font-size: 0.9em;
    }
}

@media screen and (min-width: 1200px) {
    .navigation-menu-title>p {
        font-size: 2em;
    }

    .navigation-menu-link {
        font-size: 1.1em;
    }
}

@media screen and (max-height: 800px) {

    .navigation-menu-logo,
    .navigation-menu-link,
    .navigation-menu-title {
        display: inline-block;
        height: 50px;
        line-height: 50px;
        vertical-align: middle;
    }
}

@media screen and (max-height: 2000px) {

    .navigation-menu-logo,
    .navigation-menu-link,
    .navigation-menu-title {
        display: inline-block;
        height: 70px;
        line-height: 70px;
        vertical-align: middle;
    }
}