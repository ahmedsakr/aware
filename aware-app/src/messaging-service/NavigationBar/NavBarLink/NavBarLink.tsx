import React from 'react';
import './NavBarLink.css';

type NavBarLinkProps = {
    className: string,
    icon: string,
    name: string
};

type NavBarLinkState = {};

export default class NavBarLink extends React.Component<NavBarLinkProps, NavBarLinkState> {
    render(): JSX.Element {
        const { className, icon, name } = this.props;

        return (
            <div className={className + " navigation-menu-link"}>
                <span className={icon} aria-hidden="true"></span>
                <p>{name}</p>
            </div>
        );
    }
}

type NavBarLogoProps = {
    className: string
};

type NavBarLogoState = {};

export class NavBarLogo extends React.Component<NavBarLogoProps, NavBarLogoState> {
    render(): JSX.Element {
        const { className } = this.props;

        return (
            <div className={className + " navigation-menu-logo"}>
                <p>AWARE</p>
            </div>
        );
    }
}

type NavBarTitleProps = {
    className: string,
    title: string
};

type NavBarTitleState = {};

export class NavBarTitle extends React.Component<NavBarTitleProps, NavBarTitleState> {
    render(): JSX.Element {

        const { className, title } = this.props;

        return (
            <div className={className + " navigation-menu-title"}>
                <p>{title}</p>
            </div>
        );
    }
}
