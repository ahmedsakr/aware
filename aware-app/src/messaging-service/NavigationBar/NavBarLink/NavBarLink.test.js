import React from 'react';
import { shallow } from 'enzyme';

import NavBarLink from './NavBarLink';

describe('NavBarLink', () => {
  it('should render correctly in "debug" mode', () => {

    const component = shallow(<NavBarLink src="/icons8-monitor-64.png" name="sysc 2100" debug />);
  
    expect(component).toMatchSnapshot();
  });
});