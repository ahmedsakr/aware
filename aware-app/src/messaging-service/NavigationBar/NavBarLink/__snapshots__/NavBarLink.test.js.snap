// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBarLink should render correctly in "debug" mode 1`] = `
<div
  class="undefined navigation-menu-link"
  onClick={[Function]}
>
  <span
    aria-hidden="true"
  />
  <p>
    sysc 2100
  </p>
</div>
`;
